clarifai/models/model_serving/cli/upload.py

```diff
@@ -0,0 +1,188 @@
+# Copyright 2023 Clarifai, Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Commandline interface for model upload utils."""
+import argparse
+import os
+import subprocess
+
+from clarifai.client.auth.helper import ClarifaiAuthHelper
+from clarifai.models.api import Models
+from clarifai.models.model_serving.model_config import MODEL_TYPES, get_model_config
+from clarifai.models.model_serving.model_config.inference_parameter import InferParamManager
+
+from ..utils import login
+from .base import BaseClarifaiCli
+
+
+class UploadCli(BaseClarifaiCli):
+
+  @staticmethod
+  def register(parser: argparse._SubParsersAction):
+    creator_parser = parser.add_parser("upload", help="Upload component to Clarifai platform")
+    sub_creator_parser = creator_parser.add_subparsers()
+
+    UploadModelSubCli.register(sub_creator_parser)
+
+    creator_parser.set_defaults(func=UploadCli)
+
+
+class UploadModelSubCli(BaseClarifaiCli):
+
+  @staticmethod
+  def register(parser: argparse._SubParsersAction):
+    upload_parser = parser.add_parser("model", help="Upload Clarifai model")
+    upload_parser.add_argument(
+        "--url", type=str, required=True, help="Direct download url of zip file")
+    upload_parser.add_argument("--id", type=str, required=True, help="Model ID")
+    upload_parser.add_argument(
+        "--user-app",
+        type=str,
+        required=False,
+        help="User ID and App ID separated by '/', e.g., <user_id>/<app_id>")
+    upload_parser.add_argument(
+        "--type",
+        type=str,
+        required=False,
+        choices=MODEL_TYPES,
+        default="",
+        help="Clarifai model type")
+    upload_parser.add_argument(
+        "--desc", type=str, required=False, default="", help="Short desccription of model")
+    upload_parser.add_argument(
+        "--update-version",
+        action="store_true",
+        required=False,
+        help="Update exist model with new version")
+    upload_parser.add_argument(
+        "--infer-param",
+        required=False,
+        default="",
+        help="Path to json file contains inference parameters")
```

Comment:
are we going to use json or yaml?

Reply:
We'll use `.yaml`, but this one is for `inference_paramters` used when user input args one by one (like previous). User also can use only `.yaml` with all args provided in it.